src/twisted/trial/test/test_deferred.py

```diff
@@ -210,6 +210,19 @@ def test_timeoutZero(self) -> None:
         assert isinstance(result.errors[0][1], Failure)
         self._wasTimeout(result.errors[0][1])
 
+    def test_addCleanupPassDefault(self) -> None:
+        result = self.runTest("test_addCleanupPassDefault")
+        self.assertTrue(result.wasSuccessful())
+        self.assertEqual(result.testsRun, 1)
+
+    def test_addCleanupTimeout(self) -> None:
+        result = self.runTest("test_addCleanupTimeout")
+        self.assertFalse(result.wasSuccessful())
+        self.assertEqual(result.testsRun, 1)
+        self.assertEqual(len(result.errors), 1)
+        assert isinstance(result.errors[0][1], Failure)
+        self._wasTimeout(result.errors[0][1])
```

Comment:
I see that `_wasTimeout` only checks for 

```
self.assertEqual(error.check(defer.TimeoutError), defer.TimeoutError)
```

Can we also check that "cleanup function" is part of the error message, to make sure the timeout is not accidentally triggered by someting else ?

Reply:
Fixed now.